src/ir/objc.rs

```diff
@@ -194,9 +207,61 @@ impl ObjCInterface {
             }
             CXChildVisit_Continue
         });
+
+        if interface.is_category() {
+            // If this interface is a category, we need to find the interface that this category
+            // extends.
+            if let Some(ref mut ty) =
+                Self::get_parent_ty(ctx, real_interface_id_for_category)
+            {
+                if let TypeKind::ObjCInterface(ref mut real_interface) =
+                    ty.kind_mut()
+                {
+                    if !real_interface.is_category() {
+                        real_interface.categories.push((
+                            interface.rust_name(),
+                            interface.template_names.clone(),
+                        ));
+                    }
+                }
+            }
+        }
         Some(interface)
     }
 
+    fn get_parent_ty(
+        ctx: &mut BindgenContext,
+        parent_id: Option<ItemId>,
+    ) -> Option<&mut Type> {
+        // This is pretty gross but using the ItemResolver doesn't yield a mutable reference.
+        let mut ty = ctx.resolve_item_fallible(parent_id?)?.kind().as_type()?;
+        let mut item_id: Option<&TypeId> = None;
+        loop {
+            match ty.kind() {
+                TypeKind::ResolvedTypeRef(ref_id) => {
+                    let ref_item: ItemId = ref_id.into();
+                    ty = ctx
+                        .resolve_item_fallible(ref_item)?
+                        .kind()
+                        .as_type()?;
+                    //ty = ref_item.kind().as_type()()?;;
+                    item_id = Some(ref_id);
+                }
+                TypeKind::ObjCInterface(..) => {
```

Comment:
bleh, this seems a bit backwards because it looks for the `TypeKind::ObjCInterface` but doesn't use it in the loop. It does resolve multiple layers of TypeRefs though. I kinda wish I had a test case to verify this but eh.